src/commands/sync.ts

```diff
@@ -607,56 +619,66 @@ export default class Sync extends BaseCommand<typeof Sync> {
       });
     }, this.flags["file-push-delay"]);
 
-    this.watcher
-      .add(`${this.dir}/**/*`)
-      .on("error", (error) => void this.stop(error))
-      .on("all", (event, absolutePath, stats) => {
-        const normalizedPath = this.normalize(absolutePath, event == "addDir" || event == "unlinkDir");
+    this.on("error", (error) => void this.stop(error)).on("all", (event: string, absolutePath: string, renamedPath: string) => {
+      const filePath = event === "rename" || event === "renameDir" ? renamedPath : absolutePath;
+      const isDirectory = event === "renameDir" || event === "addDir" || event === "unlinkDir";
+      const normalizedPath = this.normalize(filePath, isDirectory);
+      const stats = Sync.fileStats(filePath);
 
-        if (stats?.isSymbolicLink?.()) {
-          this.debug("skipping event caused by symlink %s", normalizedPath);
-          return;
-        }
+      // this shouldn't ever be the case using watcher since it doesn't support symlinks but we'll keep it here just in case we ever switch to another lib that does support it
+      if (stats?.isSymbolicLink?.()) {
```

Comment:
`stats` shouldn't be nullable any more so we should be able to remove the `?` here and the `assert`s below.

Reply:
`stats` can be nullable if you merge https://github.com/gadget-inc/ggt/pull/541, so you can ignore the above comment